containers/ecr-viewer/src/app/services/evaluateFhirDataService.tsx

```diff
@@ -604,6 +607,76 @@ export const evaluateEncounterData = (fhirBundle: Bundle) => {
   return evaluateData(encounterData);
 };
 
+/**
+ * Evaluates Hospital Encounter Admission and Discharge diagnosis data from the FHIR bundle and formats it into structured data for display.
+ * @param fhirBundle - The FHIR bundle containing hospital encounter data.
+ * @returns An array of evaluated and formatted hospital encounter data.
+ */
+export const evaluateHospitalEncounterData = (fhirBundle: Bundle) => {
+  const hospitalEncounterData = [
+    {
+      title: "Hospital Admission Diagnosis",
+      value: evaluateEncounterDiagnosisData(
+        fhirBundle,
+        "46241-6",
+        "Hospital Admission Diagnosis",
+      ),
+      table: true,
+    },
+    {
+      title: "Hospital Discharge Diagnosis",
+      value: evaluateEncounterDiagnosisData(
+        fhirBundle,
+        "11535-2",
+        "Hospital Discharge Diagnosis",
+      ),
+      table: true,
+    },
+  ];
+  return evaluateData(hospitalEncounterData);
+};
+
+/**
+ * Evaluates either Hospital Admission Diagnosis or Discharge diagnosis data from the FHIR bundle and formats it into structured data for display.
+ * @param fhirBundle - The FHIR bundle containing hospital encounter data.
+ * @param code - the associated code for the section to be evaluated. e.g. 46241-6 for Admission Dx or 11535-2 for Discharge Dx
+ * @param caption - A string to set the caption for the UI element
+ * @returns An array of evaluated and formatted hospital encounter data.
+ */
+export const evaluateEncounterDiagnosisData = (
+  fhirBundle: Bundle,
+  code: string,
+  caption: string,
+) => {
+  const dxRefs = evaluateAll(
+    fhirBundle,
+    fhirPathMappings.hospitalEncounterDiagnosisRef,
+    { code },
+  );
+
+  const dx: Condition[] = dxRefs
+    .map((x) => {
+      return evaluateReference<Condition>(fhirBundle, formatReference(x));
+    })
+    .filter((x): x is Condition => Boolean(x));
+
+  if (dx.length === 0) return;
+
+  const dxColumns = [
+    {
+      columnName: "Problem",
+      infoPath: "hospitalEncounterDiagnosisCode",
+    },
+    {
+      columnName: "Date/Time",
+      infoPath: "hospitalEncounterDiagnosisDateTime",
```

Comment:
these info paths need to be updated to the new names (and we should make sure tests catch that this scenario)